F/api tests

p4u opened the pull request.

p4u committed:
add helper functions to make the api tests easier

Signed-off-by: p4u <pau@dabax.net>

p4u committed:
finish transaction test and update to go 1.24

Signed-off-by: p4u <pau@dabax.net>

emmdim commented on subscriptions/subscriptions.go:
```diff
@@ -91,6 +98,15 @@ func (p *Subscriptions) HasTxPermission(
+	}
+
+	// Check if the organization has a subscription
+	if org.Subscription.PlanID == 0 {
+		return false, fmt.Errorf("organization has no subscription plan")
```
PlanID 0 is a valid plan id. The check with 0 here does not bring the result you expected

emmdim commented on subscriptions/subscriptions.go:
```diff
@@ -141,6 +157,12 @@ func (p *Subscriptions) HasDBPersmission(userEmail, orgAddress string, permissio
 	}
+
+	// Check if the organization has a subscription
+	if org.Subscription.PlanID == 0 {
```
Same as above

p4u committed:
apitest: extend transaction test to create a new process

Signed-off-by: p4u <pau@dabax.net>

p4u committed:
fix: prevent panic in subscriptions.HasTxPermission when org has no subscription

p4u committed:
refactor: update users_test.go to use the same helpers and structure as transaction_test.go

p4u committed:
apitest: add census test

Signed-off-by: p4u <pau@dabax.net>

p4u committed:
refactor: move errors to errors package

p4u committed:
split signer funder logic and new errors package

Signed-off-by: p4u <pau@dabax.net>

p4u committed:
use argon2 hashing for passwords and census data

Signed-off-by: p4u <pau@dabax.net>

p4u committed:
refactor: modify SetBulkCensusMembership to return progress channel

emmdim committed:
Updates locally stored plans

p4u committed:
api: extend the census membership add endpoint with async support

If used the query parameter ?async=true the process will be async
and a jobID token is returned.

Then the jobID progress can be checked with the /census/check/:jobID

Signed-off-by: p4u <pau@dabax.net>

p4u committed:
fix: update census test to correctly parse AddParticipantsResponse

p4u committed:
fix linter issues in multiple files

Signed-off-by: p4u <pau@dabax.net>
